source/common/router/router.cc

```diff
@@ -602,6 +602,7 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap& headers,
       headers.setEnvoyAttemptCount(attempt_count_);
     };
   }
+  callbacks_->streamInfo().setAttemptCount(attempt_count_);
```

Comment:
I'm not sure if this is the right place, isn't this just called once per request prior to retries happening? I think we'd need to set this somewhere in the retry logic?

Reply:
@snowp I am also setting in doRetry(): https://github.com/envoyproxy/envoy/pull/17583/files#diff-e3584a650478334f7373c0df344c85d499b5123f5237ce2bbabaad5ee53b899eR1609

Reply:
Not sure how I missed this, that seems right to me!